### PR TITLE
Add cats-effect module

### LINKED ISSUE #343: Add support for cats-effect
I'm trying to use pureconfig in combination with cats-effect's IO however as it stands the code to do this is a little faffsome, here is what it looks like in my codebase: 

```scala
  def loadSettings(path: Path): IO[MyConfigCaseClass] = {
    val load: IO[Either[Throwable, MyConfigCaseClass]] = IO {
      val config = loadConfig[MyConfigCaseClass](path)
      config.left.map(x => new RuntimeException("Should actually convert ConfigReaderFailures into exception here"))
    }
    load.flatMap(IO.fromEither)
  }
```

Would it be possible to add support for cats-effect, such that we could have a loadconfig that returns an IO of the config class? This would simplify usage with cats-effect, which it seems to me might be a fairly common use case given pureconfig seems to be the preferred config library within the cats ecosystem.

### PR DESCRIPTION
**Pull request by keirlawson**
Resolves #343

### CODE REVIEW

**Comment by keirlawson** [comment]
Build failing due to rvm issue seemingly unrelated to PR

**Comment by keirlawson** [comment]
Will rebase on master when build fix in https://github.com/pureconfig/pureconfig/pull/340/commits/2f60776a4deec9479f2a8043d5eb1a424b305e89 lands

**Comment by ruippeixotog** [comment]
@keirlawson the fix just got merged 👍

**Comment by keirlawson** [comment]
Rebased, awaiting review :)

File: modules/cats-effect/README.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Cats-effect module for PureConfig
```
Comment (melrief) [inline modules/cats-effect/README.md:1]: Not mandatory but we highly recommended to use tut to generate the markdown files. See what we do for the [cats module](https://github.com/pureconfig/pureconfig/tree/master/modules/cats/src/main/tut).

**Review by melrief — CHANGES_REQUESTED** [review]
Very nice! Would be LGTM but I have one comment about how to generate documentation. We usually use `tut`.

File: modules/cats-effect/README.md (hunk 1 above)
Comment (keirlawson) [reply modules/cats-effect/README.md:1]: Ok, will do so

**Comment by keirlawson** [comment]
Switched to tut for documentation as requested

**Review by melrief — APPROVED** [review]
LGTM! I like this new module, thank you @keirlawson ! I'd like to merge it if it is ready.

**Comment by keirlawson** [comment]
Yep, go ahead :)

**Comment by melrief** [comment]
Done! I will wait for the release of 0.9.0 to release this module (it should be soon). In the meanwhile, I'd like to add you to the [CONTRIBUTORS](https://github.com/pureconfig/pureconfig/blob/master/CONTRIBUTORS) file. If you agree, please make a PR with a single line with your name, username and your email and I'll merge it.

Thank you for the contribution!

**Pull request merged**